### PR TITLE
Add the `react-wrapper` package to the monorepo.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
This PR adds the functional React wrapper from https://github.com/handsontable/handsontable/pull/10831 into a separate subdirectory in the monorepo, alongside the current React wrapper.

### How has this been tested?
- Tested the new wrapper locally
- Run the new and old wrapper tests locally

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #10831 

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [x] `@handsontable/react-wrapper`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: wrappers/react-wrapper/src/helpers.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+ * @param {ReactNode} children HotTable children array.
+ * @param {String} type Either `'hot-renderer'` or `'hot-editor'`.
+ * @returns {boolean} `true` if the child of that type was found, `false` otherwise.
+ */
+function hasChildElementOfType(children: ReactNode, type: 'hot-renderer' | 'hot-editor'): boolean {
+  const childrenArray: ReactNode[] = React.Children.toArray(children);
+
+  return childrenArray.some((child) => {
+      return (child as React.ReactElement).props[type] !== void 0;
+  });
```
Comment (budnix) [inline wrappers/react-wrapper/src/helpers.tsx:132]: Can I ask to check if the linter is configured correctly? There should be 2 spaces as indentation, not 4. The `void 0` I would replace with `undefined`

File: wrappers/react-wrapper/src/helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    ownerDocument = document;
+  }
+
+  if (!bulkComponentContainer) {
+    bulkComponentContainer = ownerDocument.createDocumentFragment();
+  }
+
+  const portalContainer = cachedContainer ?? ownerDocument.createElement('DIV');
+  bulkComponentContainer.appendChild(portalContainer);
+
```
Comment (budnix) [inline wrappers/react-wrapper/src/helpers.tsx:182]:
```suggestion
  const portalContainer = cachedContainer ?? ownerDocument.createElement('div');
  
  bulkComponentContainer.appendChild(portalContainer);

```

File: wrappers/react-wrapper/src/hotColumn.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        columnSettings.editor = makeEditorClass(localEditorHooksRef, localEditorClassInstance);
+      } else if (props.hotEditor) {
+        columnSettings.editor = props.hotEditor;
+      }
+
+      return columnSettings
+    };
+
+    const columnSettings = createColumnSettings();
+    emitColumnSettings(columnSettings, columnIndex);
```
Comment (budnix) [inline wrappers/react-wrapper/src/hotColumn.tsx:79]:
```suggestion
    const columnSettings = createColumnSettings();
    
    emitColumnSettings(columnSettings, columnIndex);
```

**Review by budnix — CHANGES_REQUESTED** [review]

File: wrappers/react-wrapper/src/hotColumn.tsx (hunk 3 above)
Comment (jansiegel) [reply wrappers/react-wrapper/src/hotColumn.tsx:79]: Sadly, we don't have `lint` configuration in the React wrapper (the old one or the new one). There's an issue from 2021 for it: https://github.com/handsontable/dev-handsontable/issues/59.

Comment (budnix) [reply wrappers/react-wrapper/src/hotColumn.tsx:79]: Ok, good to know. Thanks.

**Review by budnix — APPROVED** [review]

**Review by budnix — APPROVED** [review]

**Review by budnix — APPROVED** [review]

**Pull request merged**